src/libraries/System.Net.NameResolution/src/System/Net/NameResolutionMetrics.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using System.Net.Sockets;
+
+namespace System.Net
+{
+    internal static class NameResolutionMetrics
+    {
+        private static readonly Meter s_meter = new("System.Net.NameResolution");
+
+        private static readonly Counter<long> s_lookupsRequestedCounter = s_meter.CreateCounter<long>(
+            name: "dns-lookups-requested",
```

Comment:
/cc @noahfalk to figure out the naming, since this is establishing a precedent. [Should it be](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md)   `dns.lookups.requested`?